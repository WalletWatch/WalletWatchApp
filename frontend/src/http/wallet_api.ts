import axios from 'axios';

// let host = '3f9215732d0c.vps.myjino.ru'
// let host = '127.0.0.1:8000';
let host = process.env.REACT_APP_API_URL;
let path = process.env.DJANGO_PATH

export const fetchWallet = async () => {
    const {data} = await axios.get(`http://${host}/${path}wallet/`);

    return data.data;
}

export const createWallet = async (wallet) => {
    const config = {
        headers: {
         'Content-Type': 'multipart/form-data'        
        }
    }
    
    const {data} = await axios.post(`http://${host}/${path}wallet/`, wallet, config);
    return data;
}

export const fetchOneWallet = async (id:number) => {
    const {data} = await axios.get(`http://${host}/${path}wallet/` + id + `/`);
    return data;
}

export const deleteWallet = async (id:number) => {
    const {data} = await axios.delete(`http://${host}/${path}wallet/` + id + `/`);
    return data;
}

