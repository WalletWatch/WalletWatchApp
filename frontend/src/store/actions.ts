type Token = {
    id: number;
    wallet_name: string;
    asset: string;
    asset_address: string;
    network: string;
    balance: number;
    price: number;
    updated: string;
    createdAt: string;
    wallet_id: number;
}

type Wallet = {
    id: number;
    wallet_name: string;
    wallet_address: string;
}

type Alert = {
  asset: string,
  wallet: string,
  balance: number,
  price: number,
  updated: string
}

function addWallet(wallet: Wallet) {
    return {
      type: 'ADD_WALLET',
      payload: wallet,
    };
}

function removeWallet(id: number) {
    return {
      type: 'REMOVE_WALLET',
      payload: id,
    };
}

function updateWallet(wallets: Wallet[]) {
  return {
    type: 'UPDATE_WALLET',
    payload: wallets
  }
}

function addToken(token: Token) {
    return {
      type: 'ADD_TOKEN',
      payload: token,
    };
}

function updateAllToken(tokens: Token[]) {
  return {
    type: 'UPDATE_ALL_TOKEN',
    payload: tokens,
  };
}

function updateToken(token: Token) {
  return {
    type: 'UPDATE_TOKEN',
    payload: token,
  };
}

function removeToken(id: number) {
    return {
      type: 'REMOVE_TOKEN',
      payload: id,
    };
}

function addAlert(alert: Alert) {
  return {
    type: 'ADD_ALERT',
    payload: alert
  }
}

function updateAlert(alerts: Alert[]) {
  return {
    type: 'UPDATE_ALL_ALERT',
    payload: alerts
  }
}

export { addWallet, addToken, removeToken, removeWallet, updateAllToken, updateToken, updateWallet, addAlert, updateAlert };