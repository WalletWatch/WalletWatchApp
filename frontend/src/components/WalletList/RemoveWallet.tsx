/* eslint-disable */
import React from 'react';
import { useDispatch, useSelector } from "react-redux";
import "./RemoveWallet.css";
import { removeToken, removeWallet } from "../../store/actions.ts";
import { deleteWallet } from "../../http/walletApi.ts";
import { RootState } from '../../store/store.ts';
import { deleteBalance } from '../../http/balanceApi.ts';

type Wallet = {
    id: number;
    wallet_name: string;
    wallet_address: string;
}

type RemoveWalletProps = {
    data: Wallet
}

function RemoveWallet({data}: RemoveWalletProps) {
    const dispatch = useDispatch();
    const balance = useSelector((state: RootState) => state.token);

    const handleClick = (e:any) => {
        e.preventDefault();
        balance.forEach((token) => {
            if (token.wallet_id === data.id) {
                dispatch(removeToken(token.id))
            }
        })
        deleteWallet(data.id);
        dispatch(removeWallet(data.id));
    }

    return (
        <div 
            className="remove-button-wrapper"
            onClick={handleClick}
        >
            <div className="delete_svg">
                <svg xmlns="http://www.w3.org/2000/svg" x="0px" y="0px" width="12" height="12" viewBox="0 0 24 24">
                    <path d="M 4.7070312 3.2929688 L 3.2929688 4.7070312 L 10.585938 12 L 3.2929688 19.292969 L 4.7070312 20.707031 L 12 13.414062 L 19.292969 20.707031 L 20.707031 19.292969 L 13.414062 12 L 20.707031 4.7070312 L 19.292969 3.2929688 L 12 10.585938 L 4.7070312 3.2929688 z"></path>
                </svg>
            </div>
        </div>
    );
}

export default RemoveWallet;
