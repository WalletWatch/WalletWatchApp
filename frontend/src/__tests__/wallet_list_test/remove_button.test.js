import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import RemoveWallet from '../../components/wallet_list/remove_wallet.tsx';
import { removeToken, removeWallet } from '../../store/actions.ts';
import { deleteWallet } from '../../http/wallet_api.ts';

const mockStore = configureStore([]);

jest.mock('../../http/wallet_api.ts');

describe('RemoveWallet component', () => {
    let store;
    let walletData;

    beforeEach(() => {
      store = mockStore({
        token: [
          { id: 1, wallet_id: 1 },
          { id: 2, wallet_id: 2 },
        ],
      });

      walletData = {
        id: 1,
        wallet_name: 'Test Wallet',
        wallet_address: 'test_address',
        wallet_sum: 0
      };
    });

    test('renders component correctly', () => {
      const { container } = render(
        <Provider store={store}>
          <RemoveWallet data={walletData} />
        </Provider>
      );
      expect(container).toMatchSnapshot();
    });

    test('dispatches removeToken and removeWallet actions on click', async () => {
      const dom = render(
        <Provider store={store}>
          <RemoveWallet data={walletData} />
        </Provider>
      );
      
      fireEvent.click(dom.getByTestId('remove-button-wrapper'));
      const actions = store.getActions();

      expect([actions[0]]).toEqual([
        removeToken(1), // Token id 1 is associated with wallet id 1
      ]);

      expect(actions).toContainEqual(removeWallet(walletData.id));

      expect(deleteWallet).toHaveBeenCalledWith(walletData.id);
    });
});
