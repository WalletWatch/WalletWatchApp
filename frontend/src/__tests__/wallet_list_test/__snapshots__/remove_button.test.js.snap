// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RemoveWallet component renders component correctly 1`] = `
<div>
  <div
    class="remove-button-wrapper"
    data-testid="remove-button-wrapper"
  >
    <div
      class="delete_svg"
    >
      <svg
        height="12"
        viewBox="0 0 24 24"
        width="12"
        x="0px"
        xmlns="http://www.w3.org/2000/svg"
        y="0px"
      >
        <path
          d="M 4.7070312 3.2929688 L 3.2929688 4.7070312 L 10.585938 12 L 3.2929688 19.292969 L 4.7070312 20.707031 L 12 13.414062 L 19.292969 20.707031 L 20.707031 19.292969 L 13.414062 12 L 20.707031 4.7070312 L 19.292969 3.2929688 L 12 10.585938 L 4.7070312 3.2929688 z"
        />
      </svg>
    </div>
  </div>
</div>
`;
